hasktorch/src/Torch/Typed/NN/Dropout.hs

```diff
@@ -34,9 +35,9 @@ dropoutForward ::
   IO (Tensor device dtype shape)
 dropoutForward Dropout {..} dropoutTrain = dropout dropoutProb dropoutTrain
 
-instance HasForward Dropout (Tensor device dtype shape) (Tensor device dtype shape) where
+instance HasForward Dropout (Tensor device dtype shape) where
+  type Output Dropout (Tensor device dtype shape) = Tensor device dtype shape
   forward dropout input = unsafePerformIO $ dropoutForward dropout False input
```

Comment:
alright, I guess here we are block because there isn't a dropout implementation yet that supports generators

Reply:
Yes, makes sense. My initial goal was just to get everything to compile, but makes sense that this is not very useful until we can augment the output type to make it stochastic.